[Mahesh Anugraha] iP

anuanas2007 opened the pull request:
## BRO
>"Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Bro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is
1. download it from [here](https://github.com/anuanas2007/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😎 

And it is **FREE!**

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

anuanas2007 committed:
created_Bro

anuanas2007 committed:
Add_list

anuanas2007 committed:
mark_as_done

anuanas2007 committed:
add_todo

anuanas2007 committed:
add_deadline_event

anuanas2007 committed:
added_taskcate

anuanas2007 committed:
modified_event

anuanas2007 committed:
update_bro

anuanas2007 committed:
UI_Testing

anuanas2007 committed:
Added_Exception

anuanas2007 committed:
Added_Delete

anuanas2007 committed:
Add Line

anuanas2007 committed:
Change Exp Outcome

anuanas2007 committed:
Add Storage

anuanas2007 committed:
Add DateTime

anuanas2007 committed:
Merge branch 'branch-Level-7'
Save the task

anuanas2007 committed:
Merge branch 'branch-Level-8'
Add date-time feature

anuanas2007 committed:
OOP class

anuanas2007 committed:
Add package

Samsation commented on src/main/java/bro/Parser.java:
```diff
@@ -0,0 +1,71 @@
+package bro;
+
+import bro.command.*;
```
Should this be listed explicitly?

anuanas2007 replied:
Fixed all these changes.

Samsation commented on src/main/java/bro/Parser.java:
```diff
@@ -0,0 +1,71 @@
+            }
+        }
+
+    public void checkEmptyInput(String input) throws BroException{
```
Should there be a whitespace before the '{'? I noticed the same issue in several other places too. However, this should be solved easily once you implement CheckStyle 😉

anuanas2007 replied:
sure 
Thanks, I will make all these changes!!

Samsation commented on src/main/java/bro/Storage.java:
```diff
@@ -0,0 +1,84 @@
+                    }
+                }
+                else if(s.startsWith("[D]")){
+                    String k = s.substring(6, s.indexOf(" (by")).trim();
```
Perhaps you can use a more intuitive variable name?

Samsation commented on src/main/java/bro/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+        }
+        ui.listSize(tasks);
+    }
+}
```
Hey, I like how neat and easy to follow this class is 👍

Samsation reviewed (commented):
Overall, I found your code really easy to read and follow. Just a few nits and syntactical issues to fix.

anuanas2007 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Add gradle support

zacchaeuschok commented on src/main/java/bro/Parser.java:
```diff
@@ -0,0 +1,71 @@
+        String[] in = str.split(" ");
+        String input = in[0];
+            switch (input) {
+                case "list":
```
Perhaps you can remove the indentation for case clauses to follow the Java coding standard.

anuanas2007 replied:
I have made the changes!!
Thank you

anuanas2007 replied:
Btw I requested a re-review by mistake!
sorry

zacchaeuschok commented on src/main/java/bro/command/ExitCommand.java:
```diff
@@ -0,0 +1,19 @@
+
+public class ExitCommand extends Command {
+
+    public ExitCommand(){}
```
Perhaps you might want to include a whitespace between the bracket and the brace?

anuanas2007 committed:
Set up Gradle

anuanas2007 committed:
Fix few errors

anuanas2007 committed:
Add more exception

anuanas2007 committed:
Did Unit test

anuanas2007 committed:
Create JAR file

anuanas2007 committed:
Add Java Doc

anuanas2007 committed:
Check Coding Standard

anuanas2007 committed:
Add find feature

anuanas2007 committed:
Merge branch 'branch-A-JavaDoc'

anuanas2007 committed:
Merge branch 'branch-A-CodingStandard'

anuanas2007 committed:
Merge branch 'branch-Level-9'

anuanas2007 committed:
Done CheckStyle

anuanas2007 committed:
Done GUI Tutorial

anuanas2007 committed:
Add GUI

anuanas2007 committed:
Merge branch 'branch-Level-10'

anuanas2007 committed:
Add Enums

anuanas2007 committed:
Made GUI changes

anuanas2007 committed:
Fix issues

anuanas2007 committed:
Add assertions
Class Parser and Storage sends the time details for
deadline and event task without checking whether it is empty.

Sending empty time details can break the code during parsing it to Date
format.

As a step towards avoiding empty time details to be passed in,
let's start adding assertions that ensures invalid details are not passed.

anuanas2007 committed:
Improve code quality
Class Storage has load() method which has deep nesting which makes the
code hard to understand. Parser also has a reduntant method.

Having deep nested methods and repeated code decreases the readability
of the code.

As a step towards improving the code quality, let's reduce the nesting
and try to make another method for it. Also let's remove the redundant
code.

anuanas2007 committed:
Use Stream
Filter() method of TaskList is deeply nested.

This will decrease code readability.

Let's use streams in order to increase code readability.

anuanas2007 committed:
Add more TaaskList

anuanas2007 committed:
Merge pull request #2 from anuanas2007/branch-A-Assertions

Add assertions

anuanas2007 committed:
Merge master to branch-A-CodeQuality

anuanas2007 committed:
Merge branch 'master' into branch-A-Streams

anuanas2007 committed:
Merge pull request #3 from anuanas2007/branch-A-CodeQuality

Branch a code quality

anuanas2007 committed:
Merge pull request #4 from anuanas2007/branch-A-Streams

Branch a streams

anuanas2007 committed:
Set up CI

anuanas2007 committed:
Improve Code Quality

TaskList class returns the Ui and accesses the storage to update the
modification. execute() method in Command Class only returns the
String from TaskList.

To improve the code quality and to remove weak slap.

As a step towards improving the code quality, let's acccess Ui and
modify the storage from respective Command class through execute()
method.

anuanas2007 committed:
Check duplicate task

anuanas2007 committed:
Add help command

anuanas2007 committed:
Merge branch 'branch-C-DetectDuplicates'

anuanas2007 committed:
Merge branch 'branch-C-Help'

anuanas2007 committed:
Add C-BetterSearch

anuanas2007 committed:
Fix unsuccessful build checks

anuanas2007 committed:
Better help page

anuanas2007 committed:
Change GUI

anuanas2007 committed:
Give personality

anuanas2007 committed:
Handle more errors

anuanas2007 committed:
Make minor change in java doc

anuanas2007 committed:
Add user guide

anuanas2007 committed:
New version release

anuanas2007 committed:
Modify Readme

anuanas2007 committed:
Add command summary in Readme

anuanas2007 committed:
Modify command summary

anuanas2007 committed:
Modify UG
